cmake/mrt_cmake_modules-macros.cmake

```diff
@@ -917,17 +917,17 @@ function(mrt_add_nodelet nodeletname)
 
     # get the files
     mrt_glob_files(NODELET_SOURCE_FILES_INC "${MRT_ADD_NODELET_FOLDER}/*.h" "${MRT_ADD_NODELET_FOLDER}/*.hpp"
-                   "${MRT_ADD_EXECUTABLE_FOLDER}/*.hh")
-    mrt_glob_files(NODELET_SOURCE_FILES_SRC "${MRT_ADD_NODELET_FOLDER}/*.cpp" "${MRT_ADD_EXECUTABLE_FOLDER}/*.cc")
+                   "${MRT_ADD_EXECUTABLE_FOLDER}/*.hh" "${MRT_ADD_EXECUTABLE_FOLDER}/*.cuh")
+    mrt_glob_files(NODELET_SOURCE_FILES_SRC "${MRT_ADD_NODELET_FOLDER}/*.cpp" "${MRT_ADD_EXECUTABLE_FOLDER}/*.cc" "${MRT_ADD_EXECUTABLE_FOLDER}/*.cu")
 
     # Find nodelet
-    mrt_glob_files(NODELET_CPP "${MRT_ADD_NODELET_FOLDER}/*_nodelet.cpp" "${MRT_ADD_NODELET_FOLDER}/*_nodelet.cc")
+    mrt_glob_files(NODELET_CPP "${MRT_ADD_NODELET_FOLDER}/*_nodelet.cpp" "${MRT_ADD_NODELET_FOLDER}/*_nodelet.cc" "${MRT_ADD_NODELET_FOLDER}/*_nodelet.cu")
```

Comment:
That change shouldn't be necessary. It filters out the file containing the ROS entrypoint for nodelets, and that shouldn't contain CUDA code.

If it did, it wouldn't compile in some situations, because the files are directly passed to the compile without filtering build flags not supported by NVCC. Only `mrt_add_library/executable` handles that correctly.